wgscovplot/flu.py

```diff
@@ -54,27 +68,75 @@ def read_top_references_table(basedir: Path) -> pd.DataFrame:
     )
 
 
-def get_sample_top_references(basedir: Path) -> List[SampleSegmentRef]:
-    """Find files sample.topsegments.csv for each sample
+def get_segments_ref_id(
+        segments: List,
+        sample_top_references: List[SampleSegmentRef]
+) -> Dict[str, Dict[str, str]]:
+    """
+    Return a dictionary of {sample name:{ segment: ref_id}}
+    For example:
+    'WIN_CFIA_AIV_SAMPLE_96': {'1_PB2': 'MZ171357.1', '2_PB1': 'OQ733010.1', '3_PA': 'OP377327.1', '4_HA': 'OQ733004.1',
+                              '5_NP': 'OQ694923.1', '6_NA': 'OQ733005.1', '7_M': 'OQ733006.1', '8_NS': 'OQ694932.1'}
+    """
 
-    File is located in reference_sequences/sample/sample.topsegments.csv
+    out = defaultdict(lambda: defaultdict(dict, {k: "" for k in segments}))
```

Comment:
Why not just `out = defaultdict(dict)`? Is there a reason you need to set all segments to empty string?

Reply:
Some samples in analysis may not have enough all 8 segments, just want to make sure all samples have information for all 8 segments and so it can be handled correctly with front end code! Similarly for depths, variants data with default is `[]`

Reply:
It's fairly simple to check for null/undefined values on the JS side. In most cases, you're iterating through a list of segment names to get the necessary values for the selected samples.

Reply:
In JS, use Lodash [`get`](https://lodash.com/docs/4.17.15#get) with an appropriate default value instead of `isNil` checking.